packages/page/src/Application/MessageHandler/RemovePageMessageHandler.php

```diff
@@ -28,9 +37,16 @@
  */
 final class RemovePageMessageHandler
 {
+    /**
+     * @param array<string, int> $permissions
+     */
     public function __construct(
         private PageRepositoryInterface $pageRepository,
         private DomainEventCollectorInterface $domainEventCollector,
+        private AccessControlRepositoryInterface $accessControlRepository,
+        private SystemStoreInterface $systemStore,
+        private ?Security $security,
+        private array $permissions,
```

Comment:
Unsure handle this in the application layer as its unrelated to page domain, as security is more related to UserInterface or Infrastracture Layer 🤔 .

Reply:
> Unsure handle this in the application layer as its unrelated to page domain, as security is more related to UserInterface or Infrastracture Layer 🤔 .

Yeah, I wasn’t completely sure about it either. It also feels a bit odd in the controller.